server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java

```diff
@@ -180,6 +184,43 @@ public Invocations getInvocations() {
       return completedFuture(responseBuilder.build());
    }
 
+   private CompletionStage<RestResponse> getCaches(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
+      if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
+
+      class CacheInfo {
+         public String status;
+         public String name;
+         public String type;
+         public long size;
+         public boolean simpleCache;
+         public boolean transactional;
+         public boolean persistent;
+      }
+      try {
+         Set<CacheInfo> caches = cacheManager.getCacheNames().stream().map(cacheName -> {
+            Cache<Object, Object> cache = cacheManager.getCache(cacheName);
+            CacheInfo cacheInfo = new CacheInfo();
+            cacheInfo.name = cacheName;
+            cacheInfo.type = cache.getCacheConfiguration().clustering().cacheMode().toCacheType();
+            cacheInfo.status = cache.getStatus().name();
+            cacheInfo.size = cache.size();
+            cacheInfo.simpleCache = cache.getCacheConfiguration().simpleCache();
+            cacheInfo.transactional = cache.getCacheConfiguration().transaction().transactionMode().isTransactional();
+            cacheInfo.persistent = false;
+            return cacheInfo;
+         }).collect(Collectors.toSet());
+
```

Comment:
This calculation can block, so it should be wrapped in an executor

Reply:
Are you sure this is blocking? This looks like it is just a CHM underneath.

Reply:
cache.size isn't blocking?

Reply:
Ah, yes. Sorry, I was looking at the other ones :)